osu.Game/OsuGame.cs

```diff
@@ -248,6 +248,14 @@ void setBeatmap()
                     // navigate to song select if we are not already there.
 
                     menuScreen.MakeCurrent();
+
+                    if (Beatmap.Disabled)
+                    {
+                        // we may need to wait for a lease to be returned.
+                        Schedule(() => PresentBeatmap(beatmap));
```

Comment:
Sure we don't need the same delay as in `LoadScore()` here? Seems like it used to fix visual breakage at some point (https://github.com/ppy/osu/pull/436#discussion_r104308458).